ui-core/src/main/java/com/adyen/checkout/components/ui/view/AdyenComponentView.kt

```diff
@@ -86,19 +95,35 @@ class AdyenComponentView @JvmOverloads constructor(
         configuration: Configuration,
         coroutineScope: CoroutineScope,
     ) {
-        removeAllViews()
+        binding.frameLayoutComponentContainer.removeAllViews()
         viewType ?: return
 
         val componentView = viewProvider.getView(viewType, context, attrs, defStyleAttr)
         this.componentView = componentView
 
         val localizedContext = context.createLocalizedContext(configuration.shopperLocale)
 
+        if (viewType is ButtonComponentViewType) {
+            binding.payButton.text = localizedContext.getString(viewType.buttonTextResId)
+        }
+
         val view = componentView.getView()
-        addView(view)
+        binding.frameLayoutComponentContainer.addView(view)
         view.updateLayoutParams { width = LayoutParams.MATCH_PARENT }
 
         componentView.initView(delegate, coroutineScope, localizedContext)
+
+        if (isConfirmationRequired) {
+            binding.payButton.setOnClickListener {
```

Comment:
What if the delegate changes dynamically? In that case we need to replace the click listener and set the visibility again

Reply:
It should be okay since this is called every time delegate changes. I set the listener to null in `else` condition here and hide the button.